File: app/src/main/java/com/example/chuckjokes/MainActivity.java
package com.example.chuckjokes;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.PopupMenu;
import androidx.fragment.app.Fragment;

import android.os.Bundle;

import com.example.chuckjokes.Jokes.wireframe.JokesFragment;
import com.example.chuckjokes.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    private MainPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        presenter = new MainPresenter(this);

        binding.bottomNavigation.setOnNavigationItemSelectedListener(item -> {
            switch (item.getItemId()) {
                case R.id.jokes_item:
                    presenter.jokesItemSelected();
                    break;
                case R.id.browser_item:
                    presenter.browserItemSelected();
                    break;
            }
            return true;
        });

        binding.toolbar.menuBtn.setOnClickListener(v -> {
            PopupMenu popup = new PopupMenu(this, v);
            popup.setOnMenuItemClickListener(item -> {
                //однажды тут будет ещё пара кнопок, поэтому свитч
                switch (item.getItemId()) {
                    case R.id.about:
                        presenter.onAboutClicked();
                        return true;
                    default:
                        return false;
                }
            });
            popup.inflate(R.menu.overflow_menu);
            popup.show();
        });
    }

    private void changeFragment(Fragment newFragment) {
        getSupportFragmentManager().beginTransaction().replace(binding.fragmentView.getId(), newFragment).commit();
    }

    public void setJokesFragment() {
        changeFragment(new JokesFragment());
    }

    public void setBrowserFragment() {
        changeFragment(new BrowserFragment());
    }

    public void setErrorFragment(Exception exception) {
        changeFragment(new ErrorFragment(exception));
    }

    public void setSelectedNavItem(int id) {
        binding.bottomNavigation.setSelectedItemId(id);
    }

}

File: app/src/main/java/com/example/chuckjokes/main/MainActivity.kt
package com.example.chuckjokes.main

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import androidx.appcompat.widget.PopupMenu
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import com.example.chuckjokes.R
import com.example.chuckjokes.databinding.ActivityMainBinding
import com.example.chuckjokes.util.viewModelsExt

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    private val viewModel by viewModelsExt {
        MainViewModel()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.bottomNavigation.setupWithNavController(binding.mainContainerView.getFragment<NavHostFragment>().navController)

        binding.toolbar.menuBtn.setOnClickListener { v: View ->
            val popup = PopupMenu(this, v)
            popup.setOnMenuItemClickListener { item: MenuItem ->
                when (item.itemId) {
                    R.id.about -> {
                        viewModel.onAboutClicked()
                        return@setOnMenuItemClickListener true
                    }
                    else -> return@setOnMenuItemClickListener false
                }
            }
            popup.inflate(R.menu.overflow_menu)
            popup.show()
        }

        //TODO: разобраться с навигационным стеком.
    // По переходу на фрагмент ошибки, вернутся назад к ленте нельзя,
    // нижнее меню подсвечивает не тот элемент и выйти на главный экран нельзя вовсе.

        // TODO: теряется подсветка выбранного элемента. почему так? мб из-за того что выше.
//        viewModel.selectedNavItemId.observe(this) {
//            binding.bottomNavigation.selectedItemId = it
//        }
    }
}